src/app/locationForm/number/LocationNumber.jsx

```diff
@@ -1,25 +1,113 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import { selectLocationData } from '../../../reducers';
 import { updateLocation } from '../../../actions';
 import { getLocation } from '../../../actions';
 import Form from '../common/Form';
-import FormEdit from '../common/FormEdit';
 import FormView from '../common/FormView';
 import { compose, withProps } from 'recompose';
+import PropTypes from 'prop-types';
+import Header from '../../../components/header';
+import Button from '../../../components/button';
+import Input from '../../../components/input';
 
-const LocationNumberEdit = compose(
-  withProps({
-    headerText : 'What\'s this location\'s phone number?',
-    placeholderText : 'e.g. 646-909-4591',
-  })
-)(props => <FormEdit {...props} />)
+class LocationNumberEdit extends Component {
+  constructor(props) {
+    super(props);
+
+    const [areaCode, firstThree, lastFour] = props.value.number ? props.value.number.split('.') : ['','',''];
+    this.state = {
+      areaCode, 
+      firstThree, 
+      lastFour,
+      extension : props.value.extension || ''
+    };
+
+    this.onSubmit = this.onSubmit.bind(this);
+    Object.keys(this.state).forEach(k => {
+      this[`onChange_${k}`] = this.onChange.bind(this, k);
+    });
+  }
+
+  componentWillReceiveProps(props){
```

Comment:
Here, too, once the user has started entering data, I don't think we'll want to overwrite it even if the server data changes, so componentWillReceiveProps should probably be removed.

(And if we did want the server updates to overwrite user input, probably the new getDerivedStateFromProps would help avoid all this code duplication.)

Reply:
That would also mean we probably don't need to add the `value` props to `Input`, and can just use `defaultValue`.